crates/biome_service/src/settings.rs

```diff
@@ -808,188 +880,134 @@ impl OverrideSettings {
         options
     }
 
-    pub fn override_js_globals(
+    /// Scans and uses the last override to override the `globals`
+    pub fn to_override_js_globals(
         &self,
         path: &BiomePath,
-        base_set: &Option<IndexSet<String>>,
+        base_globals: &Option<IndexSet<String>>,
     ) -> IndexSet<String> {
         self.patterns
             .iter()
             // Reverse the traversal as only the last override takes effect
             .rev()
             .find_map(|pattern| {
-                if pattern.languages.javascript.globals.is_some()
-                    && pattern.include.matches_path(path)
-                    && !pattern.exclude.matches_path(path)
-                {
+                // Early return to avoid unnecessary clone
+                pattern.languages.javascript.globals.as_ref()?;
+
+                if pattern.include.matches_path(path) && !pattern.exclude.matches_path(path) {
                     pattern.languages.javascript.globals.clone()
```

Comment:
nit: I'm not sure if i'd prefer this over a more explicit `if ... is_none()`. IMO it seemed fine the way it was before.

perhaps a `let else`? Something like:
```suggestion
                let Some(globals) = pattern.languages.javascript.globals.as_ref()? else { 
                    return None;
                };

                if pattern.include.matches_path(path) && !pattern.exclude.matches_path(path) {
                    Some(globals.clone())
```

Reply:
Clippy yells at this suggestion. So I changed it to 4a82887